### ISSUE TITLE
[Feature]: Remove hipify as a build dependency.

### ISSUE THREAD

**Issue by trixirt** [ROCmSoftwarePlatform/rccl#1465]
### Suggestion Description

hipify is only needed to run when the nccl codebase is updated.
Can hipify be moved to a maintainers options and a ci system generate/commit the hipified code ?
This would reduce the complexity and speed up the build. 

### Operating System

ALL

### GPU

ALL

### ROCm Component

RCCL

**Comment by IMbackK** [comment]
yeah i second this, its pretty silly having to use hipify at compiletime on a first party lib.

**Comment by nileshnegi** [comment]
we'll look into this hipified-code-only approach.
however, most of the build time is spent in the linking stage... let us know if you measured something different.
also, you can reduce build time by building only for local GPU architecture using `./install.sh -l`

**Comment by IMbackK** [comment]
For me this would be more about the benefit of a lessened dependency and the complexity of having to debug the hipifyed code while going back and forth between the original cuda and the generated hip.

The compile time is not a factor for me.

**Comment by ammallya** [comment]
Imported to ROCm/rocm-systems
